### PR TITLE
Add user agent and mock client to Python.

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

File: Interoperability/python/src/qsharp/clients/iqsharp.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def busy(self) -> bool:
         return self._busy
 
     def compile(self, body):
-        return self._execute(body)
+        return "E.F.G"
```
Comment (anpaz-msft) [inline Interoperability/python/src/qsharp/clients/iqsharp.py]: I think this slipped away.

File: Interoperability/python/src/qsharp/clients/mock.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+from qsharp.serialization import map_tuples, unmap_tuples
+
+## LOGGING ##
+
+import logging
+logger = logging.getLogger(__name__)
+
+## CLASSES ##
+
+class MockClient(object):
```
Comment (anpaz-msft) [inline Interoperability/python/src/qsharp/clients/mock.py]: It is probably useful to add a `logger.info` or `logger.debug` for every command.

File: Interoperability/python/src/qsharp/clients/mock.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+# -*- coding: utf-8 -*-
+##
+# mock.py: Placeholder client useful in running unit tests and in hosted CI.
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+##
+
+
+import subprocess
```
Comment (anpaz-msft) [inline Interoperability/python/src/qsharp/clients/mock.py]: most if not all of these imports are not needed.

File: Interoperability/python/src/qsharp/clients/iqsharp.py (hunk 1 above)
Comment (cgranade) [reply Interoperability/python/src/qsharp/clients/iqsharp.py]: Yes, good catch! I'll fix it, then.

File: Interoperability/python/src/qsharp/clients/mock.py (hunk 2 above)
Comment (cgranade) [reply Interoperability/python/src/qsharp/clients/mock.py]: I like that idea, yeah. I'll go on and implement it then.

File: Interoperability/python/src/qsharp/clients/mock.py (hunk 3 above)
Comment (cgranade) [reply Interoperability/python/src/qsharp/clients/mock.py]: Right, I forgot to clean that up, sorry.

**Pull request merged**